mmtrack/datasets/dancetrack_dataset.py

```diff
@@ -0,0 +1,256 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import os
+import os.path as osp
+import tempfile
+
+import motmetrics as mm
+import numpy as np
+from mmcv.utils import print_log
+from mmdet.core import eval_map
+from mmdet.datasets import DATASETS
+
+from .mot_challenge_dataset import MOTChallengeDataset
+
+try:
+    import trackeval
+except ImportError:
+    trackeval = None
+
+
+@DATASETS.register_module()
+class DanceTrackDataset(MOTChallengeDataset):
+    """Dataset for DanceTrack: https://github.com/DanceTrack/DanceTrack.
+
+    Most content is inherited from MOTChallengeDataset.
+    """
+
+    def get_dataset_cfg_for_hota(self, gt_folder, tracker_folder, seqmap):
+        """Get default configs for trackeval.datasets.MotChallenge2DBox.
+
+        Args:
+            gt_folder (str): the name of the GT folder
+            tracker_folder (str): the name of the tracker folder
+            seqmap (str): the file that contains the sequence of video names
+
+        Returns:
+            Dataset Configs for MotChallenge2DBox.
+        """
+        dataset_config = dict(
+            # Location of GT data
+            GT_FOLDER=gt_folder,
+            # Trackers location
+            TRACKERS_FOLDER=tracker_folder,
+            # Where to save eval results
+            # (if None, same as TRACKERS_FOLDER)
+            OUTPUT_FOLDER=None,
+            # Use 'track' as the default tracker
+            TRACKERS_TO_EVAL=['track'],
+            # Option values: ['pedestrian']
+            CLASSES_TO_EVAL=list(self.CLASSES),
+            # TrackEval does not support Dancetrack as an option,
+            # we use the wrapper for MOT17 dataset
+            BENCHMARK='DanceTrack',
+            # Option Values: 'train', 'val', 'test'
+            SPLIT_TO_EVAL='val',
```

Comment:
The only differences are `BENCHMARK` and `SPLIT_TO_EVAL`.
We can override the `get_benchmark()` in `MOTChanllengeDataset` to return both `DanceTrack` and `val` in order to reuse most codes as far as possiable.

Reply:
I'm not sure your intention, because in this function from `mot_challenge_dataset.py`, there is also an operation by 

```
if 'half-train' in self.ann_file:
    dataset_config[
        'GT_LOC_FORMAT'] = '{gt_folder}/{seq}/gt/gt_half-train.txt'
elif 'half-val' in self.ann_file:
    dataset_config[
        'GT_LOC_FORMAT'] = '{gt_folder}/{seq}/gt/gt_half-val.txt'
```
I have made `BENCHMARK` returned by an override `get_benchmark()` in `DanceTrackDataset` in the new commit.

Reply:
I create a [commit](https://github.com/open-mmlab/mmtracking/pull/543/commits/ae1c3c5e06dd24b49a783ed7bf9dc61c6cd69bfc) to show the intention.
The if and elif will be skipped, since dancetrack only have train and val split